openedx/core/djangoapps/programs/utils.py

```diff
@@ -673,12 +673,19 @@ def get_certificates(user, extended_program):
         certificates.append({
             'type': 'program',
             'title': extended_program['title'],
-            'url': program_credentials[0]['certificate_url'],
+            'url': get_program_certificate_url(program_credentials[0]['certificate_url']),
         })
 
     return certificates
 
 
+def get_program_certificate_url(certificate_url):
+    parsed_url = urlparse(certificate_url)
+    query_string = 'next=' + parsed_url.path
+    url_parts = (parsed_url.scheme, parsed_url.netloc, '/login/', '', query_string, '')
```

Comment:
can we use `reverse` to generate the login page url and append the proper query string to that?

Reply:
No we can't this url is of credential service. I think we can use reverse staying within the service not to other service.